dimod/discrete/discrete_quadratic_model.py

```diff
@@ -810,3 +808,315 @@ def to_numpy_vectors(self, return_offset: bool = False):
+
+        Args:
+            v: A variable in the discrete quadratic model, or a unique case
+                label.
+
+            biases (float or array-like):  If `v` is a variable, the linear
+                biases in an array.  Otherwise, the linear bias is a real
+                number.
+
+        """
+        v_k = self._unique_label_case.get(v)
+        if v_k:
+            DQM.set_linear_case(self, *v_k, biases)
+        else:
+            DQM.set_linear(self, v, biases)
+
+    def set_linear_case(self, v, case, bias):
+        """The linear bias associated with case `case` of variable `v`.
+
+        Args:
+            v: A variable in the discrete quadratic model.
+
+            case: The case of `v`.
+
+            bias (float): The linear bias.
+
+        """
+        case = self._lookup_shared_case(v, case)
+        DQM.set_linear_case(self, v, case, bias)
+
+    def set_quadratic(self, u, v, biases):
+        """Set bias(es) associated with the interaction between `u` and `v`.
+
+        Args:
+            u: A variable in the discrete quadratic model, or a unique case
+                label.  If `u` is a unique case label, `v` must be a unique
+                case label.
+
+            v: A variable in the discrete quadratic model, or a unique case
+                label.  If `u` is a unique case label, `v` must be a unique
+                case label.
+
+            biases (float or array-like/dict):
+                The quadratic bias(es).  If `u` and `v` are variables, then
+                `biases` may be a dictionary of the form
+                `{case_u, case_v: bias, ...}` or a
+                :meth:`~DiscreteQuadraticModel.num_cases(u)` by
+                :meth:`~DiscreteQuadraticModel.num_cases(v)` array-like.
+
+                If `u` and `v` are unique case labels, the quadratic bias is a
+                real number.
+
+        Raises:
+            ValueError: If `u` is a unique case label and `v` is not.
+
+        """
+        u_k = self._unique_label_case.get(u)
+        if u_k:
+            if v not in self._unique_label_case:
+                raise ValueError(f'unknown case label {v}')
+
+            v_m = self._unique_label_case[v]
+            DQM.set_quadratic_case(self, *u_k, *v_m, biases)
+        else:
+            DQM.set_quadratic(self, u, v, biases)
+
+    def set_quadratic_case(self, u, u_case, v, v_case, bias):
+        """Set the bias associated with the interaction between two cases of
+        variables `u` and `v`.
+
+        Args:
+            u: A variable in the discrete quadratic model.
+
+            u_case: The case of `u`.
+
+            v: A variable in the discrete quadratic model.
+
+            v_case: The case of `v`.
+
+            bias (float): The quadratic bias.
+
+        """
+        u_case = self._lookup_shared_case(u, u_case)
+        v_case = self._lookup_shared_case(v, v_case)
+        DQM.set_quadratic_case(self, u, u_case, v, v_case, bias)
```

Comment:
In this case is there a reason to use `DQM` rather than `super()`?

Reply:
It's explicit.

Reply:
Sure, but what does the explicitness give us in this case? Because on the down side it makes later subclassing harder and/or messes with multiple inheritance. Especially true for the one in the `__init__`.

Reply:
Explicitness reduces surprises.  But, I understand the scenarios you allude to where having used `super()` here would be desirable.  So, happy to make the change.

Reply:
Did that.